ui/pages/confirmations/components/confirm/info/shared/redesigned-gas-fees/redesigned-gas-fees.tsx

```diff
@@ -0,0 +1,333 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import React, { useEffect, useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { EtherDenomination } from '../../../../../../../../shared/constants/common';
+import { EditGasModes } from '../../../../../../../../shared/constants/gas';
+import {
+  addHexes,
+  getEthConversionFromWeiHex,
+  getValueFromWeiHex,
+  subtractHexes,
+} from '../../../../../../../../shared/modules/conversion.utils';
+import { getMinimumGasTotalInHexWei } from '../../../../../../../../shared/modules/gas.utils';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowVariant,
+} from '../../../../../../../components/app/confirm/info/row';
+import {
+  Box,
+  Button,
+  ButtonSize,
+  ButtonVariant,
+  IconName,
+  Text,
+} from '../../../../../../../components/component-library';
+import { getConversionRate } from '../../../../../../../ducks/metamask/metamask';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  IconColor,
+  JustifyContent,
+  TextAlign,
+  TextColor,
+  TextVariant,
+} from '../../../../../../../helpers/constants/design-system';
+import { useFiatFormatter } from '../../../../../../../hooks/useFiatFormatter';
+import {
+  currentConfirmationSelector,
+  getCurrentCurrency,
+} from '../../../../../../../selectors';
+import AdvancedGasFeePopover from '../../../../advanced-gas-fee-popover';
+import EditGasFeePopover from '../../../../edit-gas-fee-popover';
+import EditGasPopover from '../../../../edit-gas-popover';
+import GasTiming from '../../../../gas-timing';
+import { useSupportsEIP1559 } from '../../hooks/supports-eip-1559';
+import { useEIP1559TxFees } from '../../hooks/use-eip-1559-fees';
+import { EditGasIcon } from '../edit-gas-icon/edit-gas-icon';
+
+export const RedesignedGasFees = () => {
```

Comment:
If we're wanting this to be an isolated and reusable component, should we provide the transaction meta to it as a property, rather than forcing it to use the current confirmation?